From 3b30d40f398f3a1d9010d309617681188c2e47cc Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Tue, 6 Sep 2022 07:43:19 +0800
Subject: [PATCH 20/60] Revert "drm/vc4: hdmi: Don't access the connector state
 in reset if kmalloc fails"

This reverts commit c015d12317c503dff7fc8f20a6ae00be255cb1a3.
---
 drivers/gpu/drm/vc4/vc4_hdmi.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/vc4/vc4_hdmi.c b/drivers/gpu/drm/vc4/vc4_hdmi.c
index 978afe706ee26..9a90f22df3d20 100644
--- a/drivers/gpu/drm/vc4/vc4_hdmi.c
+++ b/drivers/gpu/drm/vc4/vc4_hdmi.c
@@ -209,9 +209,7 @@ static int vc4_hdmi_connector_get_modes(struct drm_connector *connector)
 static void vc4_hdmi_connector_reset(struct drm_connector *connector)
 {
 	drm_atomic_helper_connector_reset(connector);
-
-	if (connector->state)
-		drm_atomic_helper_connector_tv_reset(connector);
+	drm_atomic_helper_connector_tv_reset(connector);
 }
 
 static const struct drm_connector_funcs vc4_hdmi_connector_funcs = {
-- 
2.25.1

